Opcreds 3.4: fix step numbers and labels

gvargas-csa opened the pull request:
#### Summary
- [x] Fixed step numbering (Steps 10 & 11) to match the Test Plan.
Pending updates:
- [x] UpdateNOC command (Step 20 / Test Plan Step 22) → validate Constraint_error instead of MissingCsr.
- [x] CSRRequest (Step 24) → add missing IsForUpdateNOC = true field.

#### Related issues
Fix https://github.com/project-chip/connectedhomeip/issues/38404

#### Testing
Validated using darwin-x64-all-clusters-no-ble/chip-all-clusters-app

gvargas-csa committed:
Opcreds-3.4: fix step numbers and labels

andy31415 commented on src/python_testing/TC_OPCREDS_3_4.py:
```diff
@@ -107,32 +107,36 @@ def steps_TC_OPCREDS_3_4(self):
+                    10, f"TH1 sends {send_command ('ArmFailSafe')} command to the DUT with the ExpiryLengthSeconds field set to 0", None, verify_armfailsafe_response()),
                 TestStep(
-                    11, f"TH1 {send_command('UpdateNOC')} to the Node Operational Credentials cluster", None, verify_noc_response("InvalidPublicKey")),
-                TestStep(12, "TH1 generates a new Trusted Root Certificate and Private Key and saves as new_root_cert and new_root_key so that TH can generate an NOC for UpdateNOC that doesn’t chain to the original root"),
-                TestStep(13, "TH1 generates a new NOC and ICAC"),
+                    11, f"TH1 sends {send_command ('ArmFailSafe')} command to the DUT with the ExpiryLengthSeconds field set to 900", None, verify_armfailsafe_response()),
```
@gvargas-csa bot seems correct: why is there an extra space?

gvargas-csa committed:
Opcreds 3.4: validate ConstraintError instead of MissingCsr step 22

gvargas-csa committed:
Opcreds 3.4: update steps to match new test plan revisions

gvargas-csa committed:
Opcreds 3.4: Restyled by autopep8

gvargas-csa committed:
Opcreds 3.4: Align with latest test plan changes

Adding steps to close commissioning window and disarm failsafe.

gvargas-csa committed:
Opcreds 3.4: Restyled by autopep8

gvargas-csa commented on src/python_testing/TC_OPCREDS_3_4.py:
```diff
@@ -107,32 +111,42 @@ def steps_TC_OPCREDS_3_4(self):
-                    11, f"TH1 {send_command('UpdateNOC')} to the Node Operational Credentials cluster", None, verify_noc_response("InvalidPublicKey")),
-                TestStep(12, "TH1 generates a new Trusted Root Certificate and Private Key and saves as new_root_cert and new_root_key so that TH can generate an NOC for UpdateNOC that doesn’t chain to the original root"),
-                TestStep(13, "TH1 generates a new NOC and ICAC"),
+                    12, f"TH1 {send_command('CSRequest')} with the IsForUpdateNOC field set to true", None, verify_csr_not_update()),
```
Done. Thank you!

gvargas-csa commented on src/python_testing/TC_OPCREDS_3_4.py:
```diff
@@ -196,110 +210,122 @@ async def test_TC_OPCREDS_3_4(self):
+            asserts.fail("Unexpected error sending CSRRequest command")
         except InteractionModelError as e:
-            asserts.assert_equal(e.status, Status.UnsupportedAccess, "Failure status returned from UpdateNOC")
+            asserts.assert_equal(e.status, Status.InvalidCommand, "Failure status returned from CSRRequest")
```
Got it! I’ve made that change - please take a look at the updated test plan here: https://github.com/CHIP-Specifications/chip-test-plans/pull/5519.

raju-apple commented on src/python_testing/TC_OPCREDS_3_4.py:
```diff
@@ -196,110 +213,132 @@ async def test_TC_OPCREDS_3_4(self):
+
+        self.step(23)
+        cmd = opcreds.Commands.CSRRequest(CSRNonce=random.randbytes(32), isForUpdateNOC=False)
+        csr_not_update = await self.send_single_cmd(dev_ctrl=self.default_controller, node_id=self.dut_node_id, cmd=cmd)
```
Are we using this "csr_not_update" anywhere else ?

gvargas-csa committed:
Opcreds 3.4: Remove unused variable csr_not_update

CSRRequest still required before UpdateNOC command to trigger
ConstraintError instead of kMissingCsr.

gvargas-csa committed:
Opcreds 3.4: restyled by autopep8

gvargas-csa committed:
Merge branch 'master' into fix/opcreds-35-test-failure
